*** /home/user/_MYPROJECTS/TDE-github/2-tqt3/tqt/Makefile	2025-01-04 19:18:16.094118116 +0300
--- /home/user/TDE4patch/2-tqt3/tqt/Makefile	2025-01-04 20:40:22.196001689 +0300
***************
*** 11,17 ****
  install: FORCE
  	@$(MAKE) qt.install
  
! all: symlinks src-qmake src-tqmoc sub-src sub-tools
  	@echo
  	@echo "The TQt library is now built in ./lib"
  	@echo "The TQt examples are built in the directories in ./examples"
--- 11,18 ----
  install: FORCE
  	@$(MAKE) qt.install
  
! all: symlinks src-qmake src-moc sub-src sub-tools sub-tutorial sub-examples
! 
  	@echo
  	@echo "The TQt library is now built in ./lib"
  	@echo "The TQt examples are built in the directories in ./examples"
