--- qt-x11-free/mkspecs/linux-g++-64/qmake.conf~	2013-07-09 11:52:13 +0300
+++ qt-x11-free/mkspecs/linux-g++-64/qmake.conf	2013-07-09 11:52:22 +0300
@@ -45,7 +45,7 @@
 QMAKE_INCDIR_X11	= /usr/X11R6/include
 QMAKE_LIBDIR_X11	= /usr/X11R6/lib64
 QMAKE_INCDIR_QT		= $(QTDIR)/include
-QMAKE_LIBDIR_QT		= $(QTDIR)/lib64
+QMAKE_LIBDIR_QT		= $(QTDIR)/lib
 QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
 QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib64
 
