--- tqt/qmake/generators/unix/unixmake.cpp	Sun Jan  5 10:14:51 2025
+++ tqt/qmake/generators/unix/unixmake.cpp	Sun Jan  5 13:16:59 2025
@@ -837,7 +837,7 @@
 	    ret += "\n\t";
 	ret += TQString(resource ? "-$(INSTALL_DIR)" : "-$(INSTALL_FILE)") + " \"" +
 	       src_targ + "\" \"" + dst_targ + "\"";
-	if(!project->isActiveConfig("debug") && !project->isEmpty("QMAKE_STRIP") &&
+	if(false && !project->isActiveConfig("debug") && !project->isEmpty("QMAKE_STRIP") &&
 	   (project->first("TEMPLATE") != "lib" || !project->isActiveConfig("staticlib"))) {
 	    ret += "\n\t-" + var("QMAKE_STRIP");
 	    if(project->first("TEMPLATE") == "lib" && !project->isEmpty("QMAKE_STRIPFLAGS_LIB"))
